### DISCUSSION TITLE
is there a way of getting an estimate on the combined routing fees before desciding to pay a bolt11 invoice in LDK?

### DISCUSSION THREAD

**Discussion by diegovelascommerce** [lightningdevkit/rust-lightning#2440] [Help] [answered]
Hello, I'm working on a project that is integrating lighting payment into a workout app.
The client is asking is there a way to at least show what could be the estimated routing fees that the client would have to pay
in addition to the amount specified in the bolt11?

is there a way I could implement that?

and suggestions or feed back would be appreciated.  

Regards,

Diego

**Comment by tnull** [comment, marked as answer]
Yes, you could retrieve a `Route` via your `Router`'s `find_route` and then call [`Route::get_total_fees()`](https://docs.rs/lightning/latest/lightning/routing/router/struct.Route.html#method.get_total_fees) before you go ahead and pay to that route via [`ChannelManager::send_payment_with_route`](https://docs.rs/lightning/latest/lightning/ln/channelmanager/struct.ChannelManager.html#method.send_payment_with_route).

Also, while this unfortunately just slipped the upcoming 0.0.116 release, you will be able to set a max total fee when calculating a route in the future, see https://github.com/lightningdevkit/rust-lightning/issues/2339.

**Reply by diegovelascommerce** [reply]
thank you.  I'll look into it
